libats-slick/src/main/scala/com/advancedtelematic/libats/slick/db/SlickExtensions.scala

```diff
@@ -34,6 +34,11 @@ object SqlExceptions {
       case e: SQLIntegrityConstraintViolationException if e.getErrorCode == 1452 => Some(e)
       case _ => None
     }
+
+    def unapply(tuple: (Throwable, Map[String, Throwable])): Option[Throwable] =
```

Comment:
I have a feeling we are trying to mix pattern matching with a lookup in a map here. Why not use another pattern match instead of the map?